update_existing_families.py

```diff
@@ -0,0 +1,146 @@
+    for row in reader:
+        if len(row['STU_NUMBER']) > 0:
+            dp_records_multidict[row['STU_NUMBER']].append(row)
+        
+# Load district data keyed off student number ("system id" there)
+district_records_dict = {}
+with open(sys.argv[2], 'rb') as district_inputfile:
+    reader = csv.DictReader(district_inputfile)
+    validate_headers(DISTRICT_DATA_HEADERS, reader.fieldnames, "district data")
+    for row in reader:
+        district_records_dict[row['SystemID']] = row
+
+# Make updates for existing students
+dp_import_studentrecords = []
+for student_id in dp_records_multidict:
+    for dp_record in dp_records_multidict[student_id]:
+        if student_id in district_records_dict:
+            # Returning student
+            studentrecord = dp_record.copy()
+            # Update current grade
+            studentrecord['GRADE'] = district_records_dict[student_id]['Grade']
+            # Update school
+            studentrecord['SCHOOL'] = district_school_to_dp_school(district_records_dict[student_id]['School'])
+            dp_import_studentrecords.append(studentrecord)
+        elif dp_record['GRADE'] == '8' and dp_record['SCHOOL'] == 'BIS':
+            # Graduation time
+            studentrecord = dp_record.copy()
+            studentrecord['GRADE'] = '9'
+            studentrecord['SCHOOL'] = 'ALUM'
+            dp_import_studentrecords.append(studentrecord)
+
+# Build up students for each family
+students_by_family_ident = defaultdict(list)
+for key in district_records_dict:
+    family_ident = district_records_dict[key]['Family_Ident']
+    students_by_family_ident[family_ident].append(key)
+
+# Add new students for existing families
+for student_id in district_records_dict:
+    district_record = district_records_dict[student_id]
+    if student_id in dp_records_multidict:
+        continue
+    for sibling_id in students_by_family_ident[district_record['Family_Ident']]:
+        if sibling_id in dp_records_multidict:
+            siblingrecord = dp_records_multidict[sibling_id][0]
+            studentrecord = {
+                'DONOR_ID': siblingrecord['DONOR_ID'],
+                'STU_LNAME': district_record['Student Last Name'],
+                'STU_FNAME': district_record['Student First Name'],
+                'STU_NUMBER': student_id,
+                'SCHOOL': district_school_to_dp_school(district_record['School']),
+                'GRADE': district_record['Grade'],
+                'FIRST_NAME': siblingrecord['FIRST_NAME'],
+                'LAST_NAME': siblingrecord['LAST_NAME'],
+                'SP_FNAME': siblingrecord['SP_FNAME'],
+                'SP_LNAME': siblingrecord['SP_LNAME'],
+                'ADDRESS': siblingrecord['ADDRESS'],
+                'ADDRESS2': siblingrecord['ADDRESS2'],
+                'ADDRESS3': siblingrecord['ADDRESS3'],
+                'ADDRESS4': siblingrecord['ADDRESS4'],
+                'CITY': siblingrecord['CITY'],
+                'OTHER_ID': '',
```

Comment:
That will be handled by a second export file that will be imported after this one. This way I can use the existing DP address info in the matching process.